clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1249,7 +1263,7 @@ private ConsumerRecords<K, V> poll(final Timer timer, final boolean includeMetad
                     //
                     // NOTE: since the consumed position has already been updated, we must not allow
```

Comment:
This is an old documentation, I wonder if it can be a bit misleading as we don't really have a consumed position anymore. @hachikuji

Reply:
I think you could argue the comment is still correct if possibly misleading. We track the consumed position in the same field as the fetch position, but we have indeed updated it at this point. And we might end up bringing back the old field anyway.